cc_types.c

```diff
@@ -290,6 +292,69 @@ void create_struct(void)
 	i->members = last;
 }
 
+void create_enum(void)
+{
+	struct type* head = calloc(1, sizeof(struct type));
+	require(NULL != head, "Exhausted memory while creating an enum\n");
+
+	head->name = global_token->s;
+	head->type = head;
+	head->indirect = NULL;
+	head->next = global_types;
+
+	head->size = register_size; /* We treat enums as always being ints. */
+	head->is_signed = TRUE;
+
+	global_types = head;
+
+	global_token = global_token->next;
+	require_match("ERROR in create_enum\n Missing {\n", "{");
+	require(NULL != global_token, "Incomplete enum definition at end of file\n");
+
+	int next_enum_value = 0;
+	while('}' != global_token->s[0])
+	{
+		global_constant_list = sym_declare(global_token->s, NULL, global_constant_list);
+
+		global_token = global_token->next;
+		require(NULL != global_token, "Incomplete enumerator definition at end of file\n");
+		if(match("=", global_token->s))
+		{
+			global_token = global_token->next;
+			require(NULL != global_token, "Incomplete enumerator value at end of file\n");
+
+			next_enum_value = strtoint(global_token->s) + 1;
+		}
+		else
+		{
+			global_token->s = int2str(next_enum_value, 10, TRUE);
+			next_enum_value = next_enum_value + 1;
+		}
+
+		global_constant_list->arguments = global_token;
+
+		global_token = global_token->next;
+		require(NULL != global_token, "Incomplete enumerator termination at end of file\n");
+
+		if(match(";", global_token->s))
+		{
+			/* The last enumerator did not have an enumerator-expression nor a closing comma
+			 * so we overwrote the closing curly brace with the int2str value. */
+			global_token = global_token->next;
+			return;
+		}
```

Comment:
this might be fine but I'm confused when this is used/triggered?

Reply:
It's triggered on
```c
enum T {
	T_ONE
};
```

Since there's no `,` token after `T_ONE` but we still need a token to replace with the enumerator value so `}` is used instead leaving only the `;` token. If there was a trailing comma it wouldn't be triggered.